splink/database_api.py

```diff
@@ -339,3 +339,8 @@ def remove_splinkdataframe_from_cache(self, splink_dataframe: SplinkDataFrame):
 
         for k in keys_to_delete:
             del self._intermediate_table_cache[k]
+
+    def delete_tables_created_by_splink_from_db(self):
```

Comment:
Move this from linker to the database_api because otherwise it isn't accessible to methods like `profile_columns` which are using the dbapi but not the linker